src/v/cluster/members_backend.cc

```diff
@@ -566,6 +579,18 @@ ss::future<> members_backend::reallocate_replica_set(
         vassert(
           meta.new_assignment,
           "reallocation meta in reassigned state must have new_assignment");
+        if (
+          is_reconfiguring
+          && get_new_replicas(meta) != current_assignment->replicas) {
```

Comment:
Is it possible that current reconfiguration is already a cancellation? We can get an error if this is the case.